packages/dm-core-plugins/src/form/widgets/DimensionalScalarWidget.tsx

```diff
@@ -33,7 +38,7 @@ const DimensionalScalarWidget = ({
       <TextField
         unit={widgetConfig?.unit || entity?.unit}
         id={entity?.id}
-        type="number"
+        type={isNumber ? 'number' : undefined}
```

Comment:
Does it even make sense that a scalar can be anything besides "Number"? :thinking:

Reply:
Could be both strings and numbers, even though the name "scalar" implies a number, there is no need to enforce the type of the data to be number and not string. 
Works just as well as a text box.